Assets/AnkrSDK/Plugins/WalletConnectSharp.Unity/WalletConnectUnityMonoAdapter.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AnkrSDK.WalletConnectSharp.Unity.Infrastructure;
+using UnityEngine;
+
+namespace AnkrSDK.WalletConnectSharp.Unity
+{
+	public class WalletConnectUnityMonoAdapter : MonoBehaviour
+	{
+		private readonly List<IUpdatableComponent> _updatables = new List<IUpdatableComponent>();
+		private readonly List<IPausableComponent> _pausables = new List<IPausableComponent>();
+		private readonly List<IQuittableComponent> _quittables = new List<IQuittableComponent>();
+
+		private WalletConnectUnityMonoAdapter _existingInstance;
+
+		public void SetupWalletConnect(WalletConnect walletConnect)
+		{
+			_updatables.Add(walletConnect.Transport);
+			_pausables.Add(walletConnect.Transport);
+			
+			_pausables.Add(walletConnect);
+			_quittables.Add(walletConnect);
+		}
+
+		private void Awake()
+		{
+			if (_existingInstance != null)
+			{
+				Destroy(gameObject);
+			}
+			
+			DontDestroyOnLoad(gameObject);
+
+			_existingInstance = this;
+		}
+
+		private void Update()
+		{
+			foreach (var updatable in _updatables)
+			{
+				updatable.Update();
+			}
+		}
+
+		private async void OnApplicationPause(bool pauseStatus)
+		{
+			foreach (var pausable in _pausables)
+			{
+				await pausable.OnApplicationPause(pauseStatus);
```

Comment:
these should be done in parallel

Reply:
I tried it in parallel first and it did not work for some reason, probably because it has to guarantee the order of execution between Transport and WalletConnect classes

Reply:
this doesn't sound right. why do they have to be run consequently?